packages/next/server/next-server.ts

```diff
@@ -1975,6 +1975,14 @@ export default class NextNodeServer extends BaseServer {
                   ? `${parsedDestination.hostname}:${parsedDestination.port}`
                   : parsedDestination.hostname) !== req.headers.host
               ) {
+                // when we are handling a middleware prefetch and it doesn't
+                // resolve to a static data route we bail early to avoid
+                // unexpected SSR invocations
+                if (req.headers['x-middleware-prefetch']) {
+                  res.setHeader('x-middleware-skip', '1')
+                  res.body('{}').send()
```

Comment:
I'm not sure if this is an intended or unintended consequence, but this causes issues with CDNs that are configured to cache the routes `https://nextjs.org/_next/data/...`

With this change, the server would return an empty body `{}` for requests with the `x-middleware-prefetch` header which may be incidentally cached by the CDN.

For CDNs that do not allow configuring cache keys by custom headers such as Cloudflare, this empty body will be cached.

And if the user will then try to navigate to the "prefetched page", the Next.js application will throw an exception because it is then trying to load the same `https://nextjs.org/_next/data/...` URL without the `x-middleware-prefetch` header expecting the server-side props to be returned, but instead gets back `{}`.